frontend/ui/widget/textwidget.lua

```diff
@@ -46,14 +57,44 @@ function TextWidget:updateSize()
     end
     self._updated = true
 
+    -- Compute height:
+    -- Used to be:
+    --   self._height = math.ceil(self.face.size * 1.5)
+    --   self._baseline_h = self._height*0.7
+    -- But better compute baseline alignment from freetype font metrics
+    -- to get better vertical centering of text in box
+    -- (Freetype doc on this at https://www.freetype.org/freetype2/docs/tutorial/step2.html)
+    local face_height, face_ascender = self.face.ftface:getHeightAndAscender()
+    self._height = math.ceil(face_height) + 2*self.padding
+    self._baseline_h = Math.round(face_ascender) + self.padding
+    -- With our UI fonts, this usually gives 0.72 to 0.74, so text is aligned
+    -- a bit lower than before with the hardcoded 0.7
```

Comment:
Excellent!

Reply:
(Nothing excellent :) that's some previous comment that I just moved - it appears added here, but it's the same that is deleted below :)

Reply:
Oh right, I remember now. :man_facepalming: 

Still excellent though. ;-D